core/src/repair_service.rs

```diff
@@ -518,20 +527,41 @@ impl RepairService {
         slot: Slot,
         slot_meta: &SlotMeta,
         max_repairs: usize,
+        delay_threshold: Duration,
     ) -> Vec<ShredRepairType> {
         if max_repairs == 0 || slot_meta.is_full() {
             vec![]
         } else if slot_meta.consumed == slot_meta.received {
+            // check delay time of last shred
+            if slot_meta.received > 0 {
+                if let Some(shred_data) = blockstore
+                    .get_data_shred(slot, slot_meta.received - 1)
+                    .unwrap()
+                {
+                    let reference_tick =
+                        u64::from(shred::layout::get_reference_tick(&shred_data).unwrap());
```

Comment:
* rather `checked_sub` instead of `if slot_meta.received > 0` then `slot_mata.received -1`.
```rust
if let Some(index) = slot_meta.received.checked_sub(1) { 
   // ...
}
```
* are these 2 `unwrap`s here ok?
* is the extra load on blockstore due to `.get_data_shred` ok?

Reply:
- changed to `checked_sub`.
- removed `unwrap` on `get_data_shred` result. `unwrap` on reference tick matches usage in `blockstore::find_missing_indexes`.
- `get_data_shred` should not be more expensive than calling `find_missing_data_indexes`